### PR TITLE
apps/nsqd: initialize nsqd in Init(); move signal handling to go-svc

### PR DESCRIPTION
**Pull request by mreiferson**
This continues to clean up from #1331 and also fixes #1359

### CODE REVIEW

**Comment by mreiferson** [comment]
Depends on https://github.com/judwhite/go-svc/pull/28

File: apps/nsqd/main.go [hunk 1]
```diff
@@ -72,7 +62,15 @@ func (p *program) Start() error {
 	}
 	p.nsqd = nsqd
 
-	err = p.nsqd.LoadMetadata()
+	if env.IsWindowsService() {
+		dir := filepath.Dir(os.Args[0])
+		return os.Chdir(dir)
```
Comment (ploxiln) [inline apps/nsqd/main.go:67]: This used to be before `nsqd.New()` (which uses `os.Getwd()` for default datapath), also before opening the config file ... this only affects windows services, but still might want to keep their behavior consistent with before, I dunno.

Comment (mreiferson) [reply apps/nsqd/main.go:67]: Good catch

**Review by ploxiln — APPROVED** [review]
This looks good to me.

We'll need some go-svc updates for it, obviously :)

File: apps/nsqd/main.go [hunk 2]
```diff
@@ -27,21 +26,20 @@ type program struct {
...
+	if err := svc.Run(prg, syscall.SIGINT, syscall.SIGTERM); err != nil {
 		logFatal("%s", err)
 	}
 }
 
 func (p *program) Init(env svc.Environment) error {
 	if env.IsWindowsService() {
 		dir := filepath.Dir(os.Args[0])
-		return os.Chdir(dir)
+		err := os.Chdir(dir)
```
Comment (ploxiln) [inline apps/nsqd/main.go:37]: ... it turns out that go-svc (now?) has this exact code:
https://github.com/judwhite/go-svc/blob/main/svc_windows.go#L69-L76

Comment (mreiferson) [reply apps/nsqd/main.go:37]: Nice, I'll drop this

**Comment by mreiferson** [comment]
I've rebased and added acc2061, which will allow us to make progress on a build without waiting for upstream go-svc.

PTAL

**Review by ploxiln — APPROVED** [review]
LGTM

**Pull request merged**